### PR TITLE
Remove devicemapper storage driver

### PR DESCRIPTION
**Pull request by kolyshkin**
#### What type of PR is this?

/kind deprecation

#### What this PR does / why we need it:

Device mapper storage driver is hard to set up, slow, and obsoleted by overlayfs.
My best guess is nobody is using it nowadays.

Let's remove it.

#### Which issue(s) this PR fixes:

Fixes: #7002

#### Special notes for your reviewer:

* replaces #7003.
* related to https://github.com/containers/storage/pull/1622 and https://github.com/containers/storage/issues/1619.
* includes some Makefile cleanups (the first two commits).

#### Does this PR introduce a user-facing change?

```release-note
Remove device mapper storage driver.
```

### CODE REVIEW

**Comment by kolyshkin** [comment]
To add some context here, we were not able to remove DM earlier because it was used by Kata (see https://github.com/cri-o/cri-o/pull/7003#issuecomment-1640106231). This is no longer true (see https://github.com/containers/storage/pull/1622#issuecomment-2049499222).

As for "let's deprecate it first, when remove it" concern (see https://github.com/cri-o/cri-o/issues/7002#issuecomment-1572080897) -- I think there are no existing users.

**Comment by haircommander** [comment]
I vote we mark as deprecated in 1.30 and remove in 1.31, just to be safe. WDYT?

**Comment by mrunalp** [comment]
Let's loop in kata folks. I think they were using it in some scenarios.
@littlejawa @fidencio ptal

**Comment by littlejawa** [comment]
> Let's loop in kata folks. I think they were using it in some scenarios. @littlejawa @fidencio ptal

Right, we had a dependency here, but it's been removed.
Checked with @fidencio, and we're ok to remove that.

**Comment by kwilczynski** [comment]
/approve
/lgtm

Please remove the hold once everyone is OK to move forward.

/hold

**Comment by saschagrunert** [comment]
> I vote we mark as deprecated in 1.30 and remove in 1.31, just to be safe. WDYT?

Looks like we have no consensus yet.

**Comment by haircommander** [comment]
yeah if a user wants to use devicemapper, they can compile on their own still (until c/storage removes it)
/approve

WDYT @cri-o/cri-o-maintainers

**Comment by kwilczynski** [comment]
@haircommander, sounds good!

/approve
/lgtm

**Review by saschagrunert — APPROVED** [review]
/unhold

**Pull request merged**
